projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitActivity.java

```diff
@@ -348,25 +336,40 @@ public void onRequestPermissionsResult(
         permissionGranted = true;
         createOutputDir();
       } else {
-        Toast.makeText(
-                LawsuitActivity.this, "Permission is needed to create PDF.", Toast.LENGTH_LONG)
-            .show();
+        Toast.makeText(this, "Permission is needed to create PDF.", Toast.LENGTH_LONG).show();
       }
     }
   }
 
-  // Create folder in external storage for saved PDFs
-  private void createOutputDir() {
-    File directory =
-        new File(
-            Paths.get(
-                    Environment.getExternalStorageDirectory().getPath(),
-                    getString(R.string.app_name),
-                    getString(R.string.output_folder_name))
-                .toString());
-    if (!directory.exists()) {
-      directory.mkdirs();
+  @Override
+  public void onBackPressed() {
+    if (viewPager.getCurrentItem() == 0) {
+      // If the user is currently looking at the first page, allow the system to handle the
+      // Back button_blue. This calls finish() on this activity and pops the back stack.
+      super.onBackPressed();
+    } else {
+      // Otherwise, select the previous step.
+      viewPager.setCurrentItem(viewPager.getCurrentItem() - 1);
+    }
+  }
+
+  public ArrayList<Court> initCourts() {
```

Comment:
The court list should be a Prototxt, that will make this function useless

Reply:
Working on it :+1: